torch/distributed/distributed_c10d.py

```diff
@@ -25,16 +25,32 @@
     Store,
     DebugLevel,
     get_debug_level,
+    Work
```

Comment:
is this introducing the work object `dist.Work` as a top-level API, and are we sure we want to do this?